docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md

```diff
@@ -0,0 +1,383 @@
+---
+sidebar_label: "Azure Disk Encryption"
+title: "Azure Disk Encryption"
+description:
+  "How to enable disk encryption on your Azure IaaS cluster using Disk Encryption Sets with customer-managed keys"
+icon: ""
+hide_table_of_contents: false
+tags: ["public cloud", "azure", "encryption", "security"]
+sidebar_position: 12
+---
+
+Palette supports disk encryption of your Azure Kubernetes cluster using
+[Disk Encryption Sets with customer-managed keys](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#customer-managed-keys).
+
+By default, Azure encrypts all managed disks with
+[platform-managed keys](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#platform-managed-keys),
+however, customer-managed keys enable you to have greater control over your key management.
+
+You can use Disk Encryption Sets to encrypt your nodes' Operating System (OS) and data disks by selecting a preset in
+the <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" /> pack.
+
+## Limitations
+
+- Azure Disk Encryption is only supported on Azure IaaS clusters.
+
+<!-- prettier-ignore -->
+- Azure Disk Encryption is only supported when using the <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" /> pack.
+
+- If a key expires in your Key Vault, your cluster may experience operation failures. To resolve this, generate a new
+  key in Key Vault and update your Disk Encryption Set to reference the new key.
+
+  - No changes are needed in Palette, as the Palette eXtended Kubernetes pack configuration references the URI of your
+    Disk Encryption Set, which remains unchanged.
+
+  <!-- prettier-ignore -->
+  - We recommend enabling
+    [**Auto key rotation**](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#automatic-key-rotation-of-customer-managed-keys)
+    on your Disk Encryption Set so it can automatically use new key versions from your Key Vault.
+
+- Changing the Disk Encryption Set URI (`diskEncryptionSetID`) in the Palette eXtended Kubernetes pack configuration in
+  Palette will trigger a node repave.
+
+## New Cluster Profile
+
+### Prerequisites {#prerequisites-new-cluster-profile}
+
+- An Azure user account with the following roles to create the Azure Key Vault and Disk Encryption Set with the
+  necessary Key Vault access policies.
```

Comment:
Might be good to link to the Azure docs explanation of the built in key roles
https://learn.microsoft.com/en-us/azure/key-vault/general/rbac-guide?tabs=azure-cli#azure-built-in-roles-for-key-vault-data-plane-operations